realm/realm-library/src/main/java/io/realm/OrderedRealmCollection.java

```diff
@@ -35,6 +35,13 @@
     E first();
 
     /**
+     * Gets the first object from the collection. If the collection is empty, the provided default will be used.
+     *
+     * @return the first object or the provided default.
+     */
+    E first(E defaultObject);
```

Comment:
This name should match the variable name in `RealmList`, so either both should be `defaultObject` or `defaultValue`.... I probably lean towards defaultValue since these collections might also support boxed primitive types down the line.